src/charm.py

```diff
@@ -391,6 +391,11 @@ def _configure(self):  # noqa: C901
             self._stored.status["config"] = to_tuple(MaintenanceStatus("Configuring Loki"))
             return
 
```

Comment:
I would add a comment here like this:

> The config validity check does not `return` on error because if a lifecycle event
> comes in after a config change, we still want Loki to continue to function even
> with the invalid config.